### DISCUSSION TITLE
How could uninstall Flux instance without delete his CRD and other applications ?

### DISCUSSION THREAD

**Discussion by zyxie1995** [fluxcd/flux2#2897] [Q&A] [answered]
The flux uninstall command removes the Flux components and the toolkit.fluxcd.io resources from the cluster.

Imagine for a reason I don't know why, Fluxv2 don't work anymore in my cluster, I would like to uninstall Flux and install it again.
But I would like to keep all the CRD Flux in place, and keep all my applications managed by Flux running for my users.
How can I achieve that please ?
Thanks a lot !

**Comment by stefanprodan** [comment, marked as answer]
`flux uninstall` does not touches any reconciled workloads, so it’s safe to run it if you wish to reinstall Flux, docs here: https://fluxcd.io/docs/installation/#uninstall

**Reply by zyxie1995** [reply]
Hi @stefanprodan, thank you very much for your answer, yes it works, the deployment is still running after `flux uninstall`.
I would like to know please how should I do, if I want to keep all CRD installed by Flux when do uninstall flux ?

**Reply by stefanprodan** [reply]
Why would you want to keep the CRDs? You could just delete the deployments in flux-system but then the Flux custom resources will block all namespaces deletion. Uninstalling means removing the CRDs, CRs and controllers.

**Reply by zyxie1995** [reply]
Why would you want to keep the CRDs?  => I'm wondering if I could keep helmchart even I uninstall Flux. If the helmchart CRD is deleted, the helmchart resource will also be deleted then. Do you think it is necessary to keep helmchart when we need to uninstall fluxv2 ? (Actually when I reinstall flux, the helmchart will be installed again and the workloads are not be impacted)

You could just delete the deployments in flux-system but then the Flux custom resources will block all namespaces deletion. => I have tried to delete namespace `flux-system` directly by `kubectl delete ns flux-system`, but its status will be terminating (then I need to do [so](https://aws.amazon.com/fr/premiumsupport/knowledge-center/eks-terminated-namespaces/) to fix it). Do you mean the reason is that CRs in other namespace can block the `flux-system` namespace deletion ?

Uninstalling means removing the CRDs, CRs and controllers. => I understand and agree with you.

**Reply by stefanprodan** [reply]
> Do you think it is necessary to keep helmchart when we need to uninstall fluxv2 ?

No, there is no reason to do this, Flux creates proper Helm releases, that can be managed with Helm the CLI, after you delete all of Flux CRDs included.

**Reply by stefanprodan** [reply]
Also helm charts are stored by source-controller, once you delete that deployment, the actual chart is gone.

**Reply by zyxie1995** [reply]
Thanks a lot for your answer, when you said that "Flux creates proper Helm releases", you would mean Flux add a label like kustomize.toolkit.fluxcd.io/* to Helm releases resources ?

**Reply by stefanprodan** [reply]
I mean, after you uninstall Flux including the CRDs, you can manage all the releases using the Helm CLI.

**Reply by zyxie1995** [reply]
Ok, it's clear for me, thanks !
Just another question please, I have deleted the namespace `pod-info` manually (I know I should do it by changing my GitHub file, here just for a test), so the deployment of `pod-info`helm release is gone. Like you said, the helm-repo stored by source-controller is also deleted. Could you tell me why Flux delete also the helm-repo ?

**Reply by stefanprodan** [reply]
> Could you tell me why Flux delete also the helm-repo ?

It wasn't Flux, you deleted the namespace so Kubernetes deletes all objects in that namespace.

**Reply by zyxie1995** [reply]
Yes, make sense, sorry for my question.
It's clear for me now, thank you so much @stefanprodan !

**Reply by stefanprodan** [reply]
Please mark this conversation as answered.

**Reply by zyxie1995** [reply]
Yes, it's done!
Have a nice day :)
Thanks again!
